utils/parametric/spec/trace.py

```diff
@@ -154,107 +154,37 @@ def decode_v06_stats(data: bytes) -> V06StatsPayload:
+            return trace
 
 
-def find_similar_span(trace: Trace, span: Span) -> Span:
-    """Return a span from the trace which most closely matches `span`."""
+def find_span(trace: Trace, span_id: int) -> Optional[Span]:
+    """Return a span from the trace matches all fields in `span`."""
     assert len(trace) > 0
+    for span in trace:
+        if span.get("span_id") == span_id:
+            return span
 
-    max_similarity = -math.inf
-    max_similarity_span = trace[0]
-    for other_span in trace:
-        similarity = _span_similarity(span, other_span)
-        if similarity > max_similarity:
-            max_similarity = similarity
-            max_similarity_span = other_span
-    return max_similarity_span
-
-
-def find_span(trace: Trace, span: Span) -> Span:
+def find_span_by_name(trace: Trace, span_name: str) -> Optional[Span]:
     """Return a span from the trace matches all fields in `span`."""
-    return _assert_span_match(span, find_similar_span(trace, span))
-
+    assert len(trace) > 0
+    for span in trace:
+        if span.get("name") == span_name:
+            return span
 
-def find_similar_span_in_traces(traces: List[Trace], span: Span) -> Span:
-    """Return a span from the traces which most closely matches `span`."""
-    assert len(traces) > 0
 
-    max_similarity = -math.inf
-    max_similarity_span = None
-    for trace in traces:
-        similar_span = find_similar_span(trace, span)
-        if max_similarity_span is None:
-            max_similarity_span = similar_span
-        similarity = _span_similarity(span, max_similarity_span)
-        if similarity > max_similarity:
-            max_similarity_span = similar_span
-            max_similarity = similarity
-    return max_similarity_span
-
-
-def find_span_in_traces(traces: List[Trace], span: Span) -> Span:
-    """Return a span from the traces that matches all fields in `span`."""
-    return _assert_span_match(span, find_similar_span_in_traces(traces, span))
-
-
-def _assert_span_match(span: Span, similar: Span) -> Span:
-    for var in Span.__annotations__:
-        if not var.startswith("__"):
-            val = span.get(var)
-            s_val = similar.get(var)
-            if val is not None and val != s_val:
-                # TODO remove this special handling once nodejs sets service in the same way.
-                # Right now, nodejs sets a tag named "service" in meta with the correct value.
-                if var == "service":
-                    meta = similar.get("meta")
-                    if meta is not None:
-                        s_val = meta.get(var)
-                assert (
-                    val == s_val
-                ), f"Span field '{var}' mismatch '{val}' != '{s_val}'\nSpan   : {span}\nSimilar: {similar}"
-    return similar
+def find_span_by_resource(trace: Trace, resource: str) -> Optional[Span]:
+    """Return a span from the trace matches all fields in `span`."""
```

Comment:
```suggestion
    """Return a span from the trace that matches `resource`."""
```

Reply:
Honestly I am not a fan of retrieving spans by name. We will always have access to the SpanID, get spans by id is a lot more reliable than retrieving spans by names. I added this helper to avoid refactoring existing tests. On second thought I will just remove this method. We should always use `get_span()`.

Reply:
Agreed!